src/x86/sse.rs

```diff
@@ -267,6 +268,80 @@ pub unsafe fn _mm_movemask_ps(a: f32x4) -> i32 {
     movmskps(a)
 }
 
+
+/// See [`_mm_prefetch`](fn._mm_prefetch.html).
+pub const _MM_HINT_T0: i8 = 3;
+
+/// See [`_mm_prefetch`](fn._mm_prefetch.html).
+pub const _MM_HINT_T1: i8 = 2;
+
+/// See [`_mm_prefetch`](fn._mm_prefetch.html).
+pub const _MM_HINT_T2: i8 = 1;
+
+/// See [`_mm_prefetch`](fn._mm_prefetch.html).
+pub const _MM_HINT_NTA: i8 = 0;
+
+
+/// Fetch the cache line that contains address `p` using the given `strategy`.
+///
+/// The `strategy` must be one of:
+///
+/// * [`_MM_HINT_T0`](constant._MM_HINT_T0.html): Fetch into all levels of the
+///   cache hierachy.
+///
+/// * [`_MM_HINT_T1`](constant._MM_HINT_T1.html): Fetch into L2 and higher.
+///
+/// * [`_MM_HINT_T2`](constant._MM_HINT_T2.html): Fetch into L3 and higher or an
+///   implementation-specific choice (e.g., L2 if there is no L3).
+///
+/// * [`_MM_HINT_NTA`](constant._MM_HINT_NTA.html): Fetch data using the
+///   non-temporal access (NTA) hint. It may be a place closer than main memory
+///   but outside of the cache hierarchy. This is used to reduce access latency
+///   without polluting the cache.
+///
+/// The actual implementation depends on the particular CPU. This instruction
+/// is considered a hint, so the CPU is also free to simply ignore the request.
+///
+/// The amount of prefetched data depends on the cache line size of the specific
+/// CPU, but it will be at least 32 bytes.
+///
+/// Common caveats:
+///
+/// * Most modern CPUs already automatically prefetch data based on predicted
+///   access patterns.
+///
+/// * Data is usually not fetched if this would cause a TLB miss or a page
+///   fault.
+///
+/// * Too much prefetching can cause unnecessary cache evictions.
+///
+/// * Prefetching may also fail if there are not enough memory-subsystem
+///   resources (e.g., request buffers).
+///
+#[inline(always)]
+#[target_feature = "+sse"]
+#[cfg_attr(test, assert_instr(prefetcht0, strategy = _MM_HINT_T0))]
+// #[cfg_attr(test, assert_instr(prefetcht1, strategy = _MM_HINT_T1))]
+// #[cfg_attr(test, assert_instr(prefetcht2, strategy = _MM_HINT_T2))]
+// #[cfg_attr(test, assert_instr(prefetchnta, strategy = _MM_HINT_NTA))]
```

Comment:
Oh this is interesting. It looks like this is something we'd actually want to test. @alexcrichton Is there a way to have multiple `assert_instr` tests for one function?

Reply:
An interesting idea! I suspect this'd require some small tweaks to the macro definition, notably [this line](https://github.com/rust-lang-nursery/stdsimd/blob/3a8bf80ba313214274324e5231a2326067f0a5ff/stdsimd-test/assert-instr-macro/src/lib.rs#L70), we only want to append `target_feature` attributes, not all I think.

Reply:
The other change would be to allow multiple strings in case the compiler is free to choose. See https://github.com/rust-lang-nursery/stdsimd/issues/71 and https://github.com/rust-lang-nursery/stdsimd/issues/50